### PR TITLE
Add `app` directory to lintJsTrees

### LINKED ISSUE #6667: Linting tests should run on the app directory
Moving this issue to ember-cli 

https://github.com/ember-fastboot/ember-cli-fastboot/issues/307

@rwjblue can you point me in the right directions to and I will open a PR 

cheers
Dave

### PR DESCRIPTION
**Pull request by kiwiupover**
Closes #6667 and https://github.com/ember-fastboot/ember-cli-fastboot/issues/307

### CODE REVIEW

File: lib/models/addon.js [hunk 1]
```diff
@@ -1033,7 +1033,7 @@ let addonProto = {
 
     let addonJs = this.addonJsFiles(addonPath);
     let addonTemplates = this._addonTemplateFiles(addonPath);
-    let lintJsTrees = this._eachProjectAddonInvoke('lintTree', ['addon', addonJs]);
+    let lintJsTrees = this._eachProjectAddonInvoke('lintTree', ['addon', 'app', addonJs]);
```
Comment (rwjblue) [inline lib/models/addon.js]: I think you need to invoke this twice and merge the results, something like:

```js
let lintJsTrees = [
  this._eachProjectAddonInvoke('lintTree', ['addon', addonJs]);
  this._eachProjectAddonInvoke('lintTree', ['app', this.treeFor('app')])
];
```

Comment (kiwiupover) [inline lib/models/addon.js]: Ok got it thanks

**Comment by nathanhammond** [comment]
@trentmwillis @rwjblue https://travis-ci.org/ember-cli/ember-cli/jobs/191723533#L1357 seems a bit scary; don't know how easy it is to trigger the recursion.

**Comment by rwjblue** [comment]
Issue is that `treeFor('app')` calls `jshintAddonTree` [here](https://github.com/ember-cli/ember-cli/blob/master/lib/models/addon.js#L479-L481).  Work around for this is to use `this._treeFor('app')` instead (sorry for the run around @kiwiupover).

**Comment by rwjblue** [comment]
@kiwiupover - Any time to look into this?

**Comment by Turbo87** [comment]
@kiwiupover sorry, missed this PR and merged https://github.com/ember-cli/ember-cli/pull/7119 instead 😞

**Pull request closed without merging**